pytest_mock.py

```diff
@@ -288,7 +288,17 @@ def wrap_assert_methods(config):
 
 def unwrap_assert_methods():
     for patcher in _mock_module_patches:
-        patcher.stop()
+        try:
+            patcher.stop()
+        except RuntimeError as e:
+            # a patcher might have been stopped by user code (#137)
+            # so we need to catch this error here and ignore it;
+            # unfortunately there's no public API to check if a patch
+            # has been started, so catching the error it is
+            if text_type(e) == "stop called on unstarted patcher":
+                pass
+            else:
+                raise
```

Comment:
Thanks heaps for this, but I'm concerned that this might obfuscate other errors.

What would happen if some test called `.stopall()`, then some other test which relied on pytest-mock's internal mocks was called. Would it be clear what was going on?

Reply:
This might be a separate issue. It's possible to get into this situation as-is.

Reply:
Thanks for the review!

I don't think so, as we are not catching any RuntimeError, we only ignore the very specific message... as I commented above, I think this is safe, but might break in future mock releases if they ever change the message. We should also switch to a public API that allows us to check if a patcher has already been stopped, if one is ever introduced.